storage/entities/block.js

```diff
@@ -334,12 +343,36 @@ class Block extends BaseEntity {
 			parsedFilters,
 		};
 
-		return this.adapter.executeFile(
+		const result = await this.adapter.executeFile(
 			this.SQLs.select,
 			params,
 			{ expectedResultCount },
 			tx
 		);
+
+		// hasResult is always true when expectedResultCount == 1 because `executeFile` will throw an error otherwise
+		const hasResult = expectedResultCount === 1 ? true : result.length > 0;
+
+		if (parsedOptions.extended && hasResult) {
+			const transactionStorage = new Transaction(this.adapter);
+			const blockIds =
+				expectedResultCount === 1 ? [result.id] : result.map(({ id }) => id);
+			const trxFilters = { blockId_in: blockIds };
+			const trxOptions = { limit: null, extended: true };
+			const transactions = await transactionStorage.get(trxFilters, trxOptions);
+
+			if (expectedResultCount === 1) {
+				result.transactions = transactions;
+			} else {
+				result.forEach(block => {
+					block.transactions = transactions.filter(
+						({ blockId }) => blockId === block.id
+					);
+				});
+			}
+		}
+
+		return result;
```

Comment:
Why not just treat everything as collection from start and just  add condition `expectedResultCount === 1` while returning.